frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/local_image_node_widget.dart

```diff
@@ -0,0 +1,158 @@
+import 'dart:async';
+
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:flutter/material.dart';
+import 'dart:io';
+
+const String kLocalImage = 'image';
+String? folderPath;
+String? imageFile;
+String? newImage;
+
+String? copyFile(String src, String? dest, String name) {
+  try {
+    var path = File(src);
+    path.copySync('$dest/images/$name');
+    newImage = '$dest/images/$name';
+    return newImage;
+  } catch (e) {
+    debugPrint(e.toString());
+    return null;
+  }
+}
+
+Directory? checkDir(String? path) {
+  Directory dir = Directory('$path/images');
+  try {
+    if (!Directory('$path/images').existsSync()) {
+      dir.createSync(recursive: true);
+      return dir;
+    }
+    return dir;
+  } catch (e) {
+    debugPrint(e.toString());
+    return null;
+  }
+}
+
+class LocalImageNodeWidgetBuilder extends NodeWidgetBuilder {
+  LocalImageNodeWidgetBuilder({this.imageFolder});
+  final Future<Directory>? imageFolder;
+
+  @override
+  Widget build(NodeWidgetContext<Node> context) {
+    String src = context.node.attributes['image_src'].toString();
+    String imageName = context.node.attributes['name'];
+
+    imageFolder?.then((location) async {
```

Comment:
I was thinking, maybe all of the logic in here, application logic, would be cleaner if added inside a BLoC

Reply:
Do you mean adding this logic to bloc then passing the final value or just only writing the required logic in here?

Reply:
Yeah so abstracting eg. checkDir etc. To a BLoC or Cubit, and then return widget based on the state.